src/clj/harja/palvelin/raportointi/raportit/ilmoitukset.clj

```diff
@@ -0,0 +1,200 @@
+(ns harja.palvelin.raportointi.raportit.ilmoitukset
+  "Ilmoitus välilehden raportti"
+  (:require
+   [harja.palvelin.raportointi.raportit.yleinen :as yleinen :refer [rivi]]
+   [harja.domain.ely :as ely]
+   [harja.domain.tierekisteri :as tr-domain]
+   [harja.pvm :as pvm]
+   [taoensso.timbre :as log]
+   [harja.domain.tieliikenneilmoitukset :refer
+    [+ilmoitusten-selitteet+ tilan-selite] :as domain]))
+
+(defn- rivi-taulukolle
+  [tiedot lihavoi?]
+  (let [ilmoitustyyppi (domain/ilmoitustyypin-lyhenne (:ilmoitustyyppi tiedot))
+        tietojen-selitteet-koko (count (:selitteet tiedot))
+        ;; Rivittää lisätieto-kentän joka 90. kirjaimen seuraavalle riville, jos lisätiedossa paljon tekstiä 
+        lisatieto-rivitetty (clojure.string/replace (str (:lisatieto tiedot)) #"(.{90})(?!$)" "$1\n")
+        ;; Käydään läpi selitteet, lisää pilkun jos enemmän selitteitä olemassa
+        ;; "Savea tiellä, Vettä tiellä"
+        tietojen-selitteet (map (fn [x]
+                                  (let [indeksi (.indexOf (:selitteet tiedot) x)
+                                        valittaja (when (not= (+ indeksi 1) tietojen-selitteet-koko) ", ")]
+                                    (str (x +ilmoitusten-selitteet+) valittaja))) (:selitteet tiedot))]
+    (rivi
+     ;; Urakka
+     [:varillinen-teksti {:arvo (str (:urakkanimi tiedot)) :lihavoi? lihavoi?}]
+     ;; Saapunut
+     [:varillinen-teksti {:arvo (pvm/pvm-aika (:valitetty tiedot)) :lihavoi? lihavoi?}]
+     ;; Tyyppi 
+     [:varillinen-teksti {:kustomi-tyyli ilmoitustyyppi :arvo ilmoitustyyppi :lihavoi? lihavoi?}]
+     ;; Selite
+     [:varillinen-teksti {:arvo (apply str tietojen-selitteet) :lihavoi? lihavoi?}]
+     ;; Lisätieto 
+     [:varillinen-teksti {:arvo lisatieto-rivitetty :lihavoi? lihavoi?}]
+     ;; Tie 
+     [:varillinen-teksti {:arvo (tr-domain/tierekisteriosoite-tekstina (:tr tiedot) {:teksti-tie? false}) :lihavoi? lihavoi?}]
+     ;; Tila 
+     [:varillinen-teksti {:arvo ((:tila tiedot) tilan-selite) :lihavoi? lihavoi?}]
+     ;; Toimenpiteet aloitettu 
+     [:varillinen-teksti {:arvo (pvm/pvm-aika (:toimenpiteet-aloitettu tiedot)) :lihavoi? lihavoi?}])))
+
+(defn- taulukko [{:keys [otsikot tiedot]}]
+  (let [rivit (into []
+                    (remove nil?
+                            (for [x tiedot]
```

Comment:
for -> map, `(map #(rivi-taulukolle % true) tiedot)`

Reply:
Lisäksi, onko rivien aina tarkoitus olla lihavoituja?

Reply:
Korjattu, ei oikeastaan tarvi tässä lihavointia ollenkaan.
https://github.com/finnishtransportagency/harja/pull/2824/commits/0af965568c0ef1976255904a335d83f0fcc7e9b7